syntax = "proto3";

import "google/protobuf/empty.proto";

option csharp_namespace = "UserService.Protos";

service UserServiceManager {
    rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
    rpc LoginUser (LoginUserRequest) returns (LoginUserResponse);
    rpc GetUserProfile (GetUserProfileRequest) returns (GetUserProfileResponse);
    rpc Timeout (google.protobuf.Empty) returns (google.protobuf.Empty);
}

message RegisterUserRequest {
    string username = 1;
    string password = 2;
    string email = 3;
}

message RegisterUserResponse {
    string message = 1;
}

message LoginUserRequest {
    string email = 1;
    string password = 2;
}

message LoginUserResponse {
    string token = 1;
}

message GetUserProfileRequest {
    string user_id = 1;
}

message GetUserProfileResponse {
    string username = 1;
    string email = 2;
}